python/service/streaming_recognizer.py

```diff
@@ -0,0 +1,129 @@
+import threading
+from . import dictation_asr_pb2 as dictation_asr_pb2
+from . import dictation_asr_pb2_grpc as dictation_asr_pb2_grpc
+import grpc
+
+
+class RequestIterator:
+    """Thread-safe request iterator for streaming recognizer."""
+
+    def __init__(self, audio, settings):
+        # Iterator data
+        self.audio_content = audio["samples"]
+        self.settings = settings
+        self.audio_frame_rate = audio["frame_rate"]
+        frame_len = 200  # const frame len (200ms)
+        sample_width = 2
+        self.frame_samples_size = (self.audio_frame_rate // 1000) * frame_len * sample_width
+        self.request_builder = {
+            True: self._initial_request,
+            False: self._normal_request
+        }
+        # Iterator state
+        self.lock = threading.Lock()
+        self.is_initial_request = True
+        self.data_index = 0
+
+    def _initial_request(self):
+        req = StreamingRecognizer.build_configuration_request(self.audio_frame_rate, self.settings)
+        self.is_initial_request = False
+        return req
+
+    def _normal_request(self):
+        data = self.audio_content[self.data_index: (self.data_index + self.frame_samples_size)]
+        self.data_index += self.frame_samples_size
+        if self.data_index >= len(self.audio_content):
+            raise StopIteration()
+        return dictation_asr_pb2.StreamingRecognizeRequest(audio_content=data)
+
+    def __iter__(self):
+        return self
+
+    def __next__(self):
+        with self.lock:
+            return self.request_builder[self.is_initial_request]()
+
+
+class StreamingRecognizer:
+    def __init__(self, address, settings_args):
+        # Use ArgumentParser to parse settings
+        self.service = dictation_asr_pb2_grpc.SpeechStub(grpc.insecure_channel(address))
+        self.settings = settings_args
+
+    def recognize(self, audio):
+        requests_iterator = RequestIterator(audio, self.settings)
+        return self.recognize_audio_content(requests_iterator)
+
+    def recognize_audio_content(self, requests_iterator):
+        time_offsets = self.settings.time_offsets()
+
+        metadata = []
+        if self.settings.session_id():
+            metadata = [('session_id', self.settings.session_id())]
+
+        recognitions = self.service.StreamingRecognize(requests_iterator, metadata=metadata)
+
+        confirmed_results = []
+        alignment = []
+        confidence = 1.0
+
+        for recognition in recognitions:
+            if recognition.error.code:
+                print(u"Received error response: ({}) {}".format(recognition.error.code, recognition.error.message))
+            # process response type
+            elif recognition.results is not None and len(recognition.results) > 0:
```

Comment:
Here we are using just first hypothese, but user can set `--max-alternatives` in `dictation_client.py`. We should either:

- remove `max-alternatives` from configurable options
- print all hypotheses here